### PR TITLE
bug fix in catmull spline and for compiling frange.h in VS2017

### PR DESCRIPTION
**Pull request by alexxlzhou**
In frange.h, add a const to function T operator*()  in order to make it compile in VS2017.
Bug fixes in catmull_rom spline.

### CODE REVIEW

**Comment by tgeijten** [comment]
Thanks for contributing! Can you explain what the bug was in catmull_rom? It's hard to follow by looking at the commit alone.

File: xo/geometry/catmull_rom.h [hunk 1]
```diff
@@ -70,16 +70,21 @@ namespace xo
...
 				segments_.emplace_back( points[ i ], points[ i + 1 ], points[ i + 2 ], points[ i + 3 ] );
 		}
 
 		P operator()( T t ) {
+			//assume times_ is sorted
 			auto it = std::lower_bound( times_.begin(), times_.end(), t );
-			if ( it == times_.end() ) it = times_.begin();
-			if ( it == times_.end() - 1 ) it = times_.end() - 2;
-			size_t index = it - times_.begin();
-			return segments_[ index ]( ( t - *it ) / ( *( it + 1 ) - *it ) );
```
Comment (alexxlzhou) [inline xo/geometry/catmull_rom.h:82]: The index computation here is incorrect. 
The lower_bound return the iterator pointing to the first element which does not compare less than the input t.
Considering times_ as [0, 0.5, 1]:
1) if t = 0.1, the lower bound is 0.5 (instead of 0), index is 1 instead of 0;
2) if t = 1.2, it==times_.end(), the code has the index as 0 instead 1;

**Comment by tgeijten** [comment]
Thanks for the explanation and the contribution!
